Reproducible Build

mfil opened the pull request:
I'm making this pull request for further comments. I asked on IRC on Friday, and people seem to prefer that I send the commits to the mailing list before they're included.

This pull request makes builds of ovpn-dco.sys reproducible by:
- configuring the compiler and linker for reproducible builds (including submodule update)
- force-inlineing a (template) function because its instantiations appeared in a non-deterministic order
- ~~including the expected hash for ovpn-dco.sys~~

~~I couldn't build the CLI exe yet.~~

lstipakov commented:
Could you extract DMF changes into separate commit, which will become something like "Update link to DMF submodule" ?

mfil commented:
> Could you extract DMF changes into separate commit, which will become something like "Update link to DMF submodule" ?

Done!

I also fixed the path for the wpp file and added instructions for reproducible builds to the README.

I removed the file with the hash, as discussed, but when you have a release, you should write down the hash somewhere, so that people know if they got the build right. We don't want people to worry that they got a bad binary because they made a mistake!

lstipakov commented on README.md:
```diff
@@ -125,6 +125,24 @@ To see logs in attached debbuger (windbg), use `tracelog.exe` in administrator c
+4. In Powershell, navigate to the virtual drive and run `LaunchBuildEnv.cmd`.
+5. Navigate to the ovpn-dco-win project directory.
+6. Build the driver: `msbuild /p:platform=<arch> /p:configuration=release /p:signmode=off ovpn-dco-win.vcxproj /t:Build` where `<arch>` is `Win32`, `x64`, `ARM` or `ARM64`.
+7. Check that `<arch>/Release/ovpn-dco-win.sys` and the file from step 1 have the same SHA256 hash.
```
ovpn-dco-win.sys -> ovpn-dco.sys

The project is named ovpn-dco-win not to be confused with ovpn-dco project, which is Linux driver. For the file name we don't use "-win" suffix.

mfil replied:
Fixed!

I also added a bit to step 2, saying that you have to check out the correct version. Maybe that's exceedingly obvious...

Anything else you want to look at, or should I send it to the mailing list now?

lstipakov replied:
LGTM. 

In this project we accept contributions via GitHub PRs, no need to send patches to mailing list.
